### DISCUSSION TITLE
tycho 2.7 generates dependencies now in the pom of the eclipse plugin, that are not just "jars"

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-tycho/tycho#1530] [General]
hi,

we use our eclipse plugins (which for us are just jars that can also run in an OSGi environment) also in other projects that that have nothing todo with our eclipse based product..

problem now is with 2.7 that those projects don't work anymore because this happens:

```
[DEBUG]    com.servoy:servoy_shared:jar:2022.12.0.3820_rc:compile
[DEBUG]       com.google.guava:failureaccess:jar:1.0.1:compile
[DEBUG]       jakarta.servlet:jakarta.servlet-api:jar:4.0.4:compile
[DEBUG]       com.servoy:servoy_base:eclipse-plugin:2022.12.0.3820_rc:compile
[DEBUG]    com.servoy:servoy_base:jar:2022.12.0.3820_rc:compile
[DEBUG]    com.servoy:org.eclipse.dltk.javascript.rhino:jar:1.7.14.s1:compile
[DEBUG]    com.servoy:servoy_headless_client:jar:2022.12.0.3820_rc:compile
[DEBUG]       com.servoy:servoy_shared:eclipse-plugin:2022.12.0.3820_rc:compile
[DEBUG]       com.servoy:servoy_smart_client:eclipse-plugin:2022.12.0.3820_rc:compile
```

the top level are the once as a dependency:

```
 <dependencies>
                <dependency>
                    <groupId>com.servoy</groupId>
                    <artifactId>servoy_shared</artifactId>
                    <version>${project.version}</version>
			  <type>jar</type>
                </dependency>
                <dependency>
                    <groupId>com.servoy</groupId>
                    <artifactId>servoy_base</artifactId>
                    <version>${project.version}</version>
			  <type>jar</type>
                </dependency>
                <dependency>
                    <groupId>com.servoy</groupId>
                    <artifactId>org.eclipse.dltk.javascript.rhino</artifactId>
                    <version>${rhino.version}</version>
			  <type>jar</type>
                </dependency>
                <dependency>
                    <groupId>com.servoy</groupId>
                    <artifactId>servoy_headless_client</artifactId>
                    <version>${project.version}</version>
			  <type>jar</type>
                </dependency>
                <dependency>
                    <groupId>com.servoy</groupId>
                    <artifactId>servoy_smart_client</artifactId>
                    <version>${project.version}</version>
			  <type>jar</type>
                </dependency>
            </dependencies>
```

as you can see i already forced the type "jar" (i didn't have that)

But because they have dependencies to each other so mvn thinks it needs to also resolve this:

om.servoy:servoy_shared:eclipse-plugin:2022.12.0.3820_rc:compile

because in the com.servoy:servoy_headless_client.pom that is installed in my local repo i have now this (this is difference with tycho 2.6 which didn't generate this)

```
 <dependency>
      <groupId>com.servoy</groupId>
      <artifactId>servoy_shared</artifactId>
      <version>2022.12.0.3820_rc</version>
      <type>eclipse-plugin</type>
      <scope>compile</scope>
      <optional>false</optional>
    </dependency>
```

so now suddenly because of that type mvn tries to download that:

Downloading from developer_repo: https://developer.servoy.com/mvn_repository/com/servoy/servoy_shared/2022.12.0.3820_rc/servoy_shared-2022.12.0.3820_rc.eclipse-plugin

which of course doesn't work because that file is never created.. it is "jar" not "eclipse-plugin" .. 

i couldn't find anything in the release notes for this.. But how should this work now?

**Comment by laeubi** [comment]
For 2.7.x you can either disable the consumer pom: https://tycho.eclipseprojects.io/doc/latest/tycho-packaging-plugin/update-consumer-pom-mojo.html generation, or you can add tycho-maven-plugin to your build (even though it is not used) as it contains mappings from tycho-types > maven types.

**Reply by jcompagner** [reply]
thx i think just adding the tycho-maven-plugin (with extensions i notice) seems to work fine, that would be ok for me to use that in those projects..

I don't fully understand how that update-consumer because all the properties seems to have a default value false (that i would have then set to false if i read correctly)

i was afraid that i needed to add exclusions on all the dependencies.. (That would be quite huge...)

**Reply by laeubi** [reply]
You can set https://tycho.eclipseprojects.io/doc/latest/tycho-packaging-plugin/update-consumer-pom-mojo.html#skipPomGeneration explicitly to `true` **(!)**.

**Reply by jcompagner** [reply]
ah so they are not generated at all? (i did see them  in tycho 2.6, they are there, but without dependencies) not sure if i didn't want them at all..  I didn't know if they are used or not before..
